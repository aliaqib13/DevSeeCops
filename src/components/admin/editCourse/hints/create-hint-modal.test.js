import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CreateHintModal from './create-hint-modal';

describe('CreateHintModal', () => {
    let component; let
        modal;

    beforeEach(() => {
        component = shallow(<CreateHintModal />);
        modal = component.find('Modal');
    });

    it('Should render successfully', () => {
        expect(modal.exists()).toBeTruthy();
        expect(toJson(component)).toMatchSnapshot();
    });

    it('Should call addHint when modal is submitted', () => {
        const addHintMock = jest.fn(() => Promise.resolve(true));
        const toggleModalMock = jest.fn();
        component.setProps({ addHint: addHintMock, toggleModal: toggleModalMock });
        modal.props().footer[2].props.onClick();
        expect(addHintMock).toBeCalledTimes(1);
    });

    it('should push new item to hints', () => {
        expect(component.state('hintMessages').length).toEqual(1);
        component.props().footer[0].props.onClick();
        expect(component.state('hintMessages').length).toEqual(2);
    });

    it('should call "addNewHintMessage" function when press the button "Add message"', () => {
        const spy = jest.spyOn(component.instance(), 'addNewHintMessage');
        expect(spy).toBeCalledTimes(0);
        component.props().footer[0].props.onClick();
        expect(spy).toHaveBeenCalled();
    });
});
