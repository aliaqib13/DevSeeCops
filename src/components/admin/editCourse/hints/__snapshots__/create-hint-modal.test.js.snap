// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateHintModal Should render successfully 1`] = `
<Modal
  confirmLoading={false}
  footer={
    Array [
      <Button
        block={false}
        className="add-hint-btn"
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        type="link"
      >
        <Icon
          type="plus-circle"
        />
        Add message
      </Button>,
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
      >
        Cancel
      </Button>,
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        Add
      </Button>,
    ]
  }
  maskTransitionName="fade"
  okType="primary"
  title="Add Hint Message"
  transitionName="zoom"
  visible={false}
  width={520}
>
  <Collapse
    accordion={true}
    bordered={true}
    className="hint-collapse"
    defaultActiveKey="0"
    expandIconPosition="left"
    key="0"
    openAnimation={
      Object {
        "appear": [Function],
        "enter": [Function],
        "leave": [Function],
      }
    }
  >
    <CollapsePanel
      header="Hint Message 1"
      key="0"
    >
      <div
        className="block-item-hint-message"
        key="hint-0"
      >
        <CKEditor
          config={Object {}}
          data=""
          onBeforeLoad={[Function]}
          onChange={[Function]}
          readOnly={false}
          type="classic"
        />
      </div>
    </CollapsePanel>
  </Collapse>
</Modal>
`;
